chrisj117/CS-472-GROUP-2-PROJECT
# [auth] User registration verification and login

**_WHAT DOES THIS PR DO?_**
- This PR provide three endpoint; 1. registration endpoint, 2. email-verification and 3. login 
- It implements authentication using JWT 
- Send account verification emails

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

  - #88 
  - #89 
  - #93 
  - #87 
  - #91 
 
**_HOW DO I TEST OUT THIS PR?_**
- Before testing this PR, ensure you have the required environment variables which is available on the project's discord. 
- Create `.env` file at the root of `api/` directory and paste the environment variables. On your terminal type `source .env`.
 - Now, use swagger API documentation to test the endpoints: Go to `http://localhost:8000`


<img width="1009" alt="Screen Shot 2023-11-19 at 7 52 30 AM" src="https://github.com/chrisj117/CS-472-GROUP-2-PROJECT/assets/15224992/a84be8f6-116e-4dee-8218-5f7a7d618744">
